turtle_tf2_cpp/src/turtle_tf2_listener.cpp

```diff
@@ -88,38 +67,87 @@ class FrameListener : public rclcpp::Node
     std::string fromFrameRel = target_frame_.c_str();
     std::string toFrameRel = "turtle2";
 
-    geometry_msgs::msg::TransformStamped transformStamped;
-
-    // Look up for the transformation between target_frame and turtle2 frames
-    // and send velocity commands for turtle2 to reach target_frame
-    try {
-      transformStamped = tf_buffer_->lookupTransform(
-        toFrameRel, fromFrameRel,
-        tf2::TimePoint(),
-        500ms);
-    } catch (tf2::LookupException & ex) {
-      RCLCPP_INFO(this->get_logger(), "transform not ready");
-      return;
+    if (!turtle_spawned_) {
+      // Check if the service is available
+      while (!spawner_->wait_for_service(1s)) {
+        // replace if with try
```

Comment:
what's the meaning of this comment?

Reply:
I will need to update that while statement with exception handling, but I am currently not sure how.
I am following suggestions left by Chris here https://github.com/ros2/ros2_documentation/pull/1836#discussion_r689786608

Reply:
This part was updated in https://github.com/ros/geometry_tutorials/pull/53/commits/93c335c5050401afeeb73e18b3b22e485b74ae15.

But I am not 100% sure if it is the correct way to handle timer stopping.